pages/blog/slack-bot_with_cloudflare.mdx

```diff
@@ -0,0 +1,145 @@
+
+## Getting Started - Installation setup
+
+Cloudflare provides some easy and quick installation - [You can find it here](https://developers.Cloudflare.com/workers/get-started/guide/).
+
+After you install and set up all requirements, let's talk about what we did to create the Slack bot.
+
+After you installed all Cloudflare you got an index file with the following code:
+
+```ts
+export default {
+  async fetch(request) {
+    return new Response('Hello World!')
+  }
+}
+```
+
+This function will fetch any change in the URL of the worker.
+
+### What The Guild is building with Cloudflare Workers?
+
+Cloudflare also provides the option to schedule some functions. You can add this function to the index file:
+
+```ts
+async scheduled(
+    controller: ScheduledController,
+    env: Env,
+    ctx: ExecutionContext
+  ): Promise<void> {
+    await run(env);
+  }
+```
+
+The `run` function is our main function that gets the ENV secret and executes the code we create.
+
+Here you can see the `run` function, we can write any code we want to schedule in the Cloudflare Workers. We create a Slack bot that sends a daily message about all open pull requests.
+
+```ts
+export async function run(env: Env) {
+  const slack = createSlackClient(env.SLACK_TOKEN)
+  // We use Slack Client to send messages
+  const octokit = new Octokit({ auth: env.GH_BOT_TOKEN })
+  // We use Octokit to use GitHub API
+  // ... WRITE ANY CODE YOU WANT HERE
+}
+```
+
+### How can I schedule my function in the Cloudflare Worker?
+
+After you finish the installation, Cloudflare creates also `toml` file, this file contains the name, main, compatibility_date and triggers. To use the triggers, Cloudflare supports Crons and is look like that:
+
+```toml
+[triggers]
+crons = ["0 10 * * *"]
+ // This trigger will execute the function scheduled in the index
+ //  file every day at 10:00 UTC
+```
+
+### How can I fetch the data from GitHub?
+
+To get the data from GitHub we create a query-based GraphQL. GitHub provides some easy and nice explorer tools - You can use it here with the following example:
+
+```graphql
+{
+  organization(login: "the-guild-org") {
+    membersWithRole(first: 30) {
+      totalCount
+      edges {
+        node {
+          login
+          name
+        }
+      }
+    }
+  }
+}
+```
+
+After we create the query, we need to use our data. For that, we use GraphQL Codegen to generate the types based on the query.
+
+```yaml
+schema:
+  - https://api.github.com/graphql:
+      headers:
+        Authorization: Bearer ${GH_BOT_TOKEN}
+documents: 'src/**/*.ts'
+generates:
+  types.ts:
+    config:
+      scalars:
+        URI: string
+    plugins:
+      - typescript
+      - typescript-operations
+```
+
+Like the example above, we define the schema path, add Authorization to the header and after that, we generate the schema and create types to use in `run` function.
+
```

Comment:
Maybe it should include an example call to Github API with `Octokit` + `Codegen` setup?